### PR TITLE
[86bybn1q6][drag-and-drop] keyboard navigation and several minor fixes

### PR DESCRIPTION
**Pull request by msereniti**
## Motivation and Context

This PR fixes a11y issues with DragAndDrop component:
1. [x] Select looses highlight after drag and drop
1. [x] Tabline styles are broken
1. [x] Tabline items swap glitch
1. [x] Dropdonzes outline style
1. [x] Dropzone is draggable from keyboard 
1. [x] Arrows navigation not working
1. [x] In grids arrow up navigates to the prev item instead of one on the top
1. [x] Wrong VO announced positions
1. [x] Weird examples

## How has this been tested?

Manually with added examples.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Nice improve.

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly or it's not required.
- [x] Unit tests are not broken.
- [x] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

**Comment by sheila-semrush** [comment]
1. It's wrong behaviour that the dragged item is swapped with the item which "place it takes". Items shouldn't swap places, dragged item should be dropped between other items, and these other items shouldn't change order.
2. Would be nice to have at least one example using `zoneName`. We could add `zoneName` to the example with cards, or make a separate example with it.

**Review by sheila-semrush — CHANGES_REQUESTED** [review]
1. It's wrong behaviour that the dragged item is swapped with the item which "place it takes". Items shouldn't swap places, dragged item should be dropped between other items, and these other items shouldn't change order.
2. Would be nice to have at least one example using zoneName. We could add zoneName to the example with cards, or make a separate example with it.
3. Dragged element becomes gray only after changing position, it should become gray as soon as it's grabbed (or never)
image

File: semcore/drag-and-drop/src/style/drag-and-drop.shadow.css [hunk 1]
```diff
@@ -6,83 +6,85 @@ SDraggable[draggable] {
...
+SDraggable[placement='top']::before {
   top: 0;
   left: 0;
   width: 100%;
   height: 20px;
 }
 
 SDraggable[placement='top'] {
-  padding-top: var(--intergalactic-spacing-4x, 16px);
+  /* padding-top: var(--intergalactic-spacing-4x, 16px); */
```
Comment (ilyabrower) [inline semcore/drag-and-drop/src/style/drag-and-drop.shadow.css:39]: remove or uncomment

File: semcore/drag-and-drop/src/style/drag-and-drop.shadow.css [hunk 2]
```diff
@@ -6,83 +6,85 @@ SDraggable[draggable] {
...
 
-  &::after {
+  &::before {
     bottom: 0;
   }
 }
 
 SDraggable[placement='left'] {
-  padding-left: var(--intergalactic-spacing-4x, 16px);
+  /* padding-left: var(--intergalactic-spacing-4x, 16px); */
```
Comment (ilyabrower) [inline semcore/drag-and-drop/src/style/drag-and-drop.shadow.css:59]: and this

File: website/docs/.vitepress/theme/style.css [hunk 3]
```diff
@@ -494,7 +494,7 @@ video {
 }
 
 .vp-doc div[class*='language-'] {
-  overflow: auto;
+  /* overflow: auto; */
```
Comment (ilyabrower) [inline website/docs/.vitepress/theme/style.css:497]: ?

File: semcore/drag-and-drop/src/DragAndDrop.tsx [hunk 4]
```diff
@@ -224,33 +258,114 @@ class DragAndDropRoot extends Component<AsProps, {}, State> {
...
+          dragOver: null,
+          keyboardDraggingIndex: null,
+          animatedScaling: index,
+          hideHoverEffect: true,
+        });
+
+        requestAnimationFrame(() => {
+          const prevIndexNode = this.state.items[prevIndex]?.node;
+          if (prevIndexNode !== document.activeElement) return;
+          this.state.items[dragOver as any]?.node.focus();
```
Comment (ilyabrower) [inline semcore/drag-and-drop/src/DragAndDrop.tsx:318]: are you sure about `as any` ? looks like we could use `number` as array index

**Review by ilyabrower — APPROVED** [review]

**Pull request merged**